### PR TITLE
Special cards

### PR DESCRIPTION
**Pull request by Earving8**
Adds special cards! Please check carefully, it is possible that it can add many bugs

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -137,21 +150,68 @@ function useCard()
...
+        }
+        
         players[gameTurn].playerDeck.playCard(cardIndex);
-        gameTurn++;
+        
+        if(cardBeingPlayed.color == 'Special'){
+            if(cardBeingPlayed.value == 0){
+                cardWild();
+            }else if(cardBeingPlayed.value == 1){
+                cardDraw4();
```
Comment (CompSciLauren) [inline index.js]: This card has a bug. When used, the number on the screen is -1 and no card is valid to play except another one of these cards. Image: https://pasteboard.co/I8TAXG4.png

Comment (CompSciLauren) [inline index.js:181]: Issues I found:
- Next player has no card that is valid to play except another special card (when valid card is attempted to be played, it says Valid Move, but then another popup says the card does not stack)
- If the next player tries to draw a new hand, the screen immediately switches back to other player's hand.
- Does not make the next user draw 4 cards

File: index.js [hunk 2]
```diff
@@ -137,21 +150,68 @@ function useCard()
...
+                }
+                
+        }
+        
         players[gameTurn].playerDeck.playCard(cardIndex);
-        gameTurn++;
+        
+        if(cardBeingPlayed.color == 'Special'){
+            if(cardBeingPlayed.value == 0){
+                cardWild();
```
Comment (CompSciLauren) [inline index.js:179]: Issues I found (same as first 2 bullet points I wrote on line 175):
- Next player has no card that is valid to play except another special card (when valid card is attempted to be played, it says Valid Move, but then another popup says the card does not stack)
- If the next player tries to draw a new hand, the screen immediately switches back to other player's hand.

File: index.js [hunk 3]
```diff
@@ -137,21 +150,68 @@ function useCard()
...
+            }else if(cardBeingPlayed.value == 1){
+                cardDraw4();
+            }
+        }else if(cardBeingPlayed.value == 10){
+            cardDraw2();
```
Comment (CompSciLauren) [inline index.js:184]: This doesn't make a player draw 2 cards, only change seems to be playfield card becomes "[color] 10" (like expected)

File: index.js [hunk 4]
```diff
@@ -137,21 +150,68 @@ function useCard()
...
+        }else if(cardBeingPlayed.value == 11){
+            cardReverse();
```
Comment (CompSciLauren) [inline index.js:186]: This is hard to test with only 2 players, but didn't notice anything wrong with the code. We should probably test this again when we have 3 players.

File: index.js [hunk 5]
```diff
@@ -220,9 +276,63 @@ function player(deck, id, index)
...
+
+//Skips the next player in rotation
+function cardSkip(){
+    rotatePlayers();
+}
+
+function cardWild(){
+    newColor = prompt("Enter the Color You want to switch to");
+    playFieldCard.color = newColor;
+    playFieldCard.value = -1;
```
Comment (CompSciLauren) [inline index.js:328]: Why do we set it to -1 again? Is that just to make it easier on us for now?

Edit: Earving said in discord it's so you can only play a card of that color or another special.

File: index.js [hunk 6]
```diff
@@ -137,21 +150,68 @@ function useCard()
...
+                    alert("Card chosen Doesn't stack");
+                    return;
+                }
+                
+        }
+        
         players[gameTurn].playerDeck.playCard(cardIndex);
-        gameTurn++;
+        
+        if(cardBeingPlayed.color == 'Special'){
```
Comment (CompSciLauren) [inline index.js:177]: If player chooses a special 0 or 1 card, and leaves the uno tab before picking a color, then upon returning to the uno game the popup to choose color is gone and there is no color set for the current round, just shows -1 for the card value.

**Review by CompSciLauren — CHANGES_REQUESTED** [review]
Looking good! I left a few comments that should be addressed before this gets merged.

File: index.js (hunk 1 above)
Comment (TravisHull) [reply index.js:181]: Issue is caused because card color is case-sensitive

File: index.js (hunk 2 above)
Comment (TravisHull) [reply index.js:179]: Same issue as first

File: index.js (hunk 3 above)
Comment (TravisHull) [reply index.js:184]: cardDraw2() working

File: index.js (hunk 4 above)
Comment (TravisHull) [reply index.js:186]: will test when we expand to 3-4 players

File: index.js (hunk 6 above)
Comment (TravisHull) [reply index.js:177]: Could not recreate issue, leaving tab and returning works on my end.

**Review by TravisHull — APPROVED** [review]
Branch looks ready to merge

**Review by CompSciLauren — APPROVED** [review]

**Pull request merged**